File: app/modules/drawCard.js
/**
 * Draw n number of cards per each N trials in a standard deck.
 */
const R = require('r-script');

/**
 * Returns a list containing the result of the experiment both with and without replacement.
 * 
 * @module
 * @param {Number} N A positive integer that represents the N number of trials.
 * @param {Number} n A positive integer that represents the n number of cards to draw.
 */
const drawCard = (N, n) => {
    let result = {
        trials: N,
        draws: n
    };

    // Generate Experiment with replacement.
    result.w_replacement = R('./scripts/draw-card.R')
    .data(N, n, true)
    .callSync();

    // Generate Experiment without replacment
    result.wo_replacement = R('./scripts/draw-card.R')
    .data(N, n, false)
    .callSync();

    return result;
};

module.exports = drawCard;

File: app/scripts/draw-card.R
draw.card <- function(N, n, replace) {
    result = list()
    for (i in 1:N) {
        drawn.cards = sample(1:52, n, replace=as.logical(replace))
        result[[i]] = drawn.cards
    }

    return (result)
}

do.call(draw.card, input)